### DISCUSSION TITLE
How to filter through searchString using operators like 'gt', 'lt'?

### DISCUSSION THREAD

**Discussion by skelawsky** [prisma/prisma#21166] [Q&A]
Do I need to write my own parser for this or did I miss something in the documentation? I can't find info about it.

**Comment by ludralph** [comment]
Hi @skelawsky 👋 

In Prisma, you can use operators like [gt](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#gt) (greater than) and [lt](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#lt) (less than) to filter data. However, these operators are typically used with numerical values, not with a string search. For example, you can use `gt` to find all posts where the number of likes is greater than a certain value:
```js
const getPosts = await prisma.post.findMany({
  where: {
    likes: {
      gte: 9,
    },
  },
})
```
For string searches, Prisma provides the [contains](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#contains), [startsWith](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#startswith), and [endsWith](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#endswith) operators. For example, you can use 'contains' to find all posts where the title contains a certain string:

```js
const result = await prisma.post.count({
  where: {
    content: {
      contains: 'databases',
    },
  },
})
```

If you want to perform a [full-text search](https://www.prisma.io/docs/concepts/components/prisma-client/full-text-search), Prisma also supports that feature for PostgreSQL and MySQL databases. You can use the `search` operator to find all posts where the title contains certain words:
```js
// All posts that contain the word 'cat'.
const result = await prisma.posts.findMany({
  where: {
    body: {
      search: 'cat',
    },
  },
})
```

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by zebapy** [reply]
Prisma _is_ producing gt, lt, etc operators for string values. How do those work?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
